DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -1,244 +1,297 @@
+                    ap_uint<TrackBitWidths::kRinvSize> rInv,
+                    ap_uint<TrackBitWidths::kPhiSize> phi0,
+                    ap_uint<TrackBitWidths::kTanlSize> tanl,
+                    ap_uint<TrackBitWidths::kZ0Size> z0,
+                    ap_uint<TrackBitWidths::kD0Size> d0,
+                    ap_uint<TrackBitWidths::kChi2RPhiSize> chi2RPhi,  // would be total chisq if chi2Z is zero
+                    ap_uint<TrackBitWidths::kChi2RZSize> chi2RZ,
+                    ap_uint<TrackBitWidths::kBendChi2Size> bendChi2,
+                    ap_uint<TrackBitWidths::kHitPatternSize> hitPattern,
+                    ap_uint<TrackBitWidths::kMVAQualitySize> mvaQuality,
+                    ap_uint<TrackBitWidths::kMVAOtherSize> mvaOther);
 
-  unsigned int get_hitPattern();
-  unsigned int get_ispare();  // will need a converter(s) when spare bits are defined
-
-  unsigned int get_RinvBits();
-  unsigned int get_phiBits();
-  unsigned int get_tanlBits();
-  unsigned int get_z0Bits();
-  unsigned int get_d0Bits();
-  unsigned int get_chi2XYBits();
-  unsigned int get_chi2ZBits();
-  unsigned int get_BendChi2Bits();
-
-  // copy constructor
-
-  TTTrack_TrackWord(const TTTrack_TrackWord& word) {
-    initialize();
-
-    iRinv = word.iRinv;
-    iphi = word.iphi;
-    itanl = word.itanl;
-    iz0 = word.iz0;
-    id0 = word.id0;
-    ichi2XY = word.ichi2XY;
-    ichi2Z = word.ichi2Z;
-    iBendChi2 = word.iBendChi2;
-    ispare = word.ispare;
-    iHitPattern = word.iHitPattern;
-
-    // three 32-bit packed words
-    TrackWord1 = word.TrackWord1;
-    TrackWord2 = word.TrackWord2;
-    TrackWord3 = word.TrackWord3;
+private:
+  // ----------private member functions --------------
+  unsigned int digitizeSignedValue(double value, unsigned int nBits, double lsb) const {
+    unsigned int digitized_value = std::floor(std::abs(value) / lsb);
+    unsigned int digitized_maximum = (1 << (nBits - 1)) - 1;  // The remove 1 bit from nBits to account for the sign
+    if (digitized_value > digitized_maximum)
+      digitized_value = digitized_maximum;
+    if (value < 0)
+      digitized_value = (1 << nBits) - digitized_value;  // two's complement encoding
+    return digitized_value;
   }
 
-  TTTrack_TrackWord& operator=(const TTTrack_TrackWord& word) {
-    initialize();
-    iRinv = word.iRinv;
-    iphi = word.iphi;
-    itanl = word.itanl;
-    iz0 = word.iz0;
-    id0 = word.id0;
-    ichi2XY = word.ichi2XY;
-    ichi2Z = word.ichi2Z;
-    iBendChi2 = word.iBendChi2;
-    ispare = word.ispare;
-    iHitPattern = word.iHitPattern;
-
-    // three 32-bit packed words
-    TrackWord1 = word.TrackWord1;
-    TrackWord2 = word.TrackWord2;
-    TrackWord3 = word.TrackWord3;
+  unsigned int getBin(double value, const std::vector<double>& bins) const {
+    auto low = std::lower_bound(bins.begin(), bins.end(), value);
+    if (low == bins.end())
+      return (bins.size() - 1);
+    else
```

Comment:
With the current implementation, the `if (low == bins.end())` branch could be removed: `bins.end() - bins.begin()` should be the same as `bins.size()` anyway.